api.go

```diff
@@ -0,0 +1,1997 @@
+// Package ecs provides a client for ECS Extension.
+package ecs
+
+import (
+	"io"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awsutil"
+	"github.com/aws/aws-sdk-go/aws/request"
+	"github.com/aws/aws-sdk-go/private/protocol"
+	"github.com/aws/aws-sdk-go/private/protocol/restxml"
+	"github.com/aws/aws-sdk-go/service/s3"
+)
+
+type S3 struct {
+	*s3.S3
+}
+
+func New(s *s3.S3) *S3 {
+	return &S3{s}
+}
+
+var initRequest func(*request.Request)
+
+func init() {
+	initRequest = defaultInitRequestFn
+}
+
+// newRequest creates a new request for a S3 operation and runs any
+// custom request initialization.
+func (c *S3) newRequest(op *request.Operation, params, data interface{}) *request.Request {
+	req := c.NewRequest(op, params, data)
+
+	// Run custom request initialization if present
+	if initRequest != nil {
+		initRequest(req)
+	}
+
+	return req
+}
+
+const opCreateBucket = "CreateBucket"
+
+// CreateBucketExtRequest generates a request.Request
+func (c *S3) CreateBucketExtRequest(input *CreateBucketInput) (req *request.Request, output *s3.CreateBucketOutput) {
```

Comment:
accepted. will change it to "Extension". unnecessary to add ecs keyword because the github URL includes emcecs.